libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -151,19 +155,44 @@ string IRGenerator::generateFunction(FunctionDefinition const& _function)
 
 string IRGenerator::constructorCode(ContractDefinition const& _contract)
 {
-	// TODO initialize state variables in base to derived order.
-	// TODO base constructors
-	// TODO callValueCheck if there is no constructor.
+	using boost::adaptors::reverse;
+
 	if (FunctionDefinition const* constructor = _contract.constructor())
 	{
 		string out;
 		if (!constructor->isPayable())
 			out = callValueCheck();
+
 		solUnimplementedAssert(constructor->parameters().empty(), "");
-		return move(out) + m_context.functionName(*constructor) + "()\n";
+		solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
+
+		// Initialization of state variables in base-to-derived order.
+		for (ContractDefinition const* contract: reverse(_contract.annotation().linearizedBaseContracts))
+			for (VariableDeclaration const* variable: contract->stateVariables())
+				out += initializeStateVariable(*variable);
+
+		// TODO base constructors
+
+		out += m_context.functionName(*constructor) + "()\n";
+
+		return out;
 	}
+	else
+		return callValueCheck(); // callValueCheck if there is no constructor.
+}
+
+string IRGenerator::initializeStateVariable(VariableDeclaration const& _varDecl)
```

Comment:
Maybe it would be better to add that as a public function to IRGeneratorForStatements - because it makes a lot of assumptions about the code that are more related to code generation for statements than high-level code generation.